python/tvm/meta_schedule/runner/local_runner.py

```
@@ -0,0 +1,332 @@
+    ) -> List[float]:
+        f_alloc_argument: LocalRunner.T_ALLOC_ARGUMENT = get_global_func_with_default_on_worker(
+            _f_alloc_argument, default_alloc_argument
+        )
+        f_run_evaluator: LocalRunner.T_RUN_EVALUATOR = get_global_func_with_default_on_worker(
+            _f_run_evaluator, default_run_evaluator
+        )
+        f_cleanup: LocalRunner.T_CLEANUP = get_global_func_with_default_on_worker(
+            _f_cleanup, default_cleanup
+        )
+
+        @contextmanager
+        def resource_handler():
+            try:
+                yield
+            finally:
+                # Step 5. Clean up
+                f_cleanup()
+
+        with resource_handler():
+            # Step 1: create the local runtime module
+            rt_mod = tvm.runtime.load_module(artifact_path)
+            # Step 2: create the local device
+            device = tvm.runtime.device(dev_type=device_type, dev_id=0)
+            # Step 3: Allocate input arguments
+            repeated_args: List[T_ARGUMENT_LIST] = f_alloc_argument(
+                device,
+                args_info,
+                alloc_repeat,
+            )
+            # Step 4: Run time_evaluator
+            costs: List[float] = f_run_evaluator(
+                rt_mod,
+                device,
+                evaluator_config,
+                repeated_args,
+            )
+        return costs
+
+
+def default_alloc_argument(
+    device: Device,
+    args_info: T_ARG_INFO_JSON_OBJ_LIST,
+    alloc_repeat: int,
+) -> List[T_ARGUMENT_LIST]:
+    """Default function to allocate the arguments
+
+    Parameters
+    ----------
+    device: Device
+        The device to allocate the arguments
+    args_info: T_ARG_INFO_JSON_OBJ_LIST
+        The arguments info
+    alloc_repeat: int
+        The number of times to repeat the allocation
+
+    Returns
+    -------
+    repeated_args: List[T_ARGUMENT_LIST]
+        The allocation args
+    """
+    try:
```

Use the get global function method in utils.py and apply it in sanity checks so that it could fail early if the method doesn’t exist

---

Should we be using `get_global_func_with_default_on_worker` on `LocalRunner`? Right now in `LocalRunner` I didn't add `_sanity_check`. If that is required. I can add it for `LocalRunner` within this PR.

---

Oh then let's do a sanity check then. Thanks for the explanation!

---

sanity check is added.